menas/src/main/scala/za/co/absa/enceladus/menas/services/SchemaRegistryService.scala

```diff
@@ -0,0 +1,85 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.services
+
+import java.net.URL
+
+import com.typesafe.config.ConfigFactory
+import org.springframework.beans.factory.annotation.{Autowired, Value}
+import org.springframework.stereotype.Service
+import za.co.absa.enceladus.menas.controllers.SchemaController
+import za.co.absa.enceladus.menas.models.rest.exceptions.RemoteSchemaRetrievalException
+import za.co.absa.enceladus.menas.services.SchemaRegistryService.SchemaResponse
+import za.co.absa.enceladus.menas.utils.SchemaType
+import za.co.absa.enceladus.utils.config.SecureConfig
+
+import scala.io.Source
+import scala.util.control.NonFatal
+
+@Service
+class SchemaRegistryService @Autowired()() {
+
+  @Value("${menas.schemaRegistryBaseUrl}")
+  val schemaRegistryBaseUrl: String = ""
+
+  private val config = ConfigFactory.load()
+  SecureConfig.setKeyStoreProperties(config)
+  SecureConfig.setTrustStoreProperties(config)
+
+  /**
+   * Loading the latest schema by a topicName (e.g. topic1-value), the url is constructed based on the [[schemaRegistryBaseUrl]]
+   * @param topicName topic name
+   * @return
+   */
+  def loadSchemaByTopicName(topicName: String): SchemaResponse = {
+    val schemaUrl = SchemaRegistryService.getLatestSchemaUrl(schemaRegistryBaseUrl, topicName)
+    loadSchemaByUrl(schemaUrl)
+  }
+
+  /**
+   * Loading the schema by a full URL
+   * @param remoteUrl
+   * @return
+   */
+  def loadSchemaByUrl(remoteUrl: String): SchemaResponse = {
+    try {
+      val url = new URL(remoteUrl)
+      val connection = url.openConnection()
+      val mimeType = SchemaController.avscContentType // only AVSC is expected to come from the schema registry
+      val fileStream = Source.fromInputStream(connection.getInputStream)
+      val fileContent = fileStream.mkString
+      fileStream.close()
+
+      SchemaResponse(fileContent, mimeType, url)
+    } catch {
+      case NonFatal(e) =>
+        throw RemoteSchemaRetrievalException(SchemaType.Avro, s"Could not retrieve a schema file from $remoteUrl. " +
+          s"Please check the correctness of the URL and a presence of the schema at the mentioned endpoint", e)
+    }
+  }
+
+}
+
+object SchemaRegistryService {
+
+  import za.co.absa.enceladus.utils.implicits.StringImplicits._
+
+  def getLatestSchemaUrl(baseUrl: String, topicName: String): String =
+    baseUrl / "subjects" / topicName / "versions/latest/schema"
```

Comment:
Just leaving a comment about terminology, since I got a notification about this issue.
I think you confused topic with subject. Topic is a Kafka concept, while Subject is a Schema Registry concept. See also: https://docs.confluent.io/current/schema-registry/schema_registry_tutorial.html#terminology-review
The API reference uses "subjectName" and not "topicName" https://docs.confluent.io/current/schema-registry/develop/api.html#get--subjects-(string-%20subject)-versions-(versionId-%20version)-schema

In the Post request, the param is named "topicStem" which should probably be "subjectStem"

Reply:
Yes. You are completely correct and this is quite eye-opening :)

I will address this as well after I deal with https://github.com/AbsaOSS/enceladus/pull/1405#discussion_r443436688

Reply:
The terminology has been changed to mention schema registry _subject_. Thank you.